onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -261,12 +261,16 @@ def enketo(self, request, *args, **kwargs):
         elif(isinstance(self.object, Instance)):
             if request.user.has_perm("change_xform", self.object.xform):
                 return_url = request.query_params.get('return_url')
+                form_url = get_form_url(
+                    request,
+                    self.object.xform.user.username,
+                    xform_pk=self.object.xform.id)
                 if not return_url:
                     raise ParseError(_(u"return_url not provided."))
 
                 try:
-                    data["url"] = get_enketo_edit_url(
-                        request, self.object, return_url)
+                    data = enketo_url(form_url, self.object.xform.id_string)
```

Comment:
Why are we assigning the result of `enketo_url` to `data` and not `data["url"]`?

Reply:
This url is unpacked elsewhere